DFC.Digital/DFC.Digital.Service.CourseSearch/Converters/MessageConverter.cs

```diff
@@ -110,14 +128,33 @@ internal static CourseDetails ConvertToCourseDetails(this CourseDetailOutput api
                     LearnerSatisfaction = apiCourseDetail.Provider.FEChoices_LearnerSatisfaction,
                     EmployerSatisfaction = apiCourseDetail.Provider.FEChoices_EmployerSatisfaction
                 },
-                StartDateLabel = apiCourseDetail.Opportunity.FirstOrDefault()?.StartDate.Item,
+                Oppurtunities = GetOppurtunities(apiCourseDetail, activeOppurtunity.OpportunityId),
+                CourseUrl = apiCourseDetail.Course.URL,
                 CourseId = apiCourseDetail.Course.CourseID,
-                AttendanceMode = apiCourseDetail.Opportunity.FirstOrDefault()?.AttendanceMode,
-                AttendancePattern = apiCourseDetail.Opportunity.FirstOrDefault()?.AttendancePattern,
-                QualificationLevel = apiCourseDetail.Course.QualificationLevel,
-                StudyMode = apiCourseDetail.Opportunity.FirstOrDefault()?.StudyMode,
-                Duration = $"{apiCourseDetail.Opportunity.FirstOrDefault()?.Duration?.DurationValue} {apiCourseDetail.Opportunity.FirstOrDefault()?.Duration?.DurationUnit}"
+                Cost = activeOppurtunity.Price,
+                StartDateLabel = activeOppurtunity.StartDate.Item,
```

Comment:
Potential `NullReferenceException` because of code block `82 - 85`

Reply:
Updated